src/test/java/net/sf/jabref/importer/fileformat/BibteXMLImporterTest.java

```diff
@@ -0,0 +1,185 @@
+package net.sf.jabref.importer.fileformat;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.importer.OutputPrinterToNull;
+import net.sf.jabref.model.entry.BibEntry;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BibteXMLImporterTest {
+
+    @BeforeClass
+    public static void setUp() {
+        Globals.prefs = JabRefPreferences.getInstance();
+    }
+
+    @Test
+    public void testExceptionOnInputStream() throws IOException {
+        try (InputStream is = Mockito.mock(InputStream.class)) {
+            Mockito.doThrow(new IOException()).when(is).read();
+
+            BibteXMLImporter importer = new BibteXMLImporter();
+            importer.importEntries(is, new OutputPrinterToNull());
+        }
+    }
+
+    @Test
+    public void testGetFormatName() {
+        BibteXMLImporter importer = new BibteXMLImporter();
+        Assert.assertEquals("BibTeXML", importer.getFormatName());
+    }
+
+    @Test
+    public void testGetCLIId() {
+        BibteXMLImporter importer = new BibteXMLImporter();
+        Assert.assertEquals("bibtexml", importer.getCLIId());
+    }
+
+    @Test
+    public void testIsRecognizedFormatAccept() throws IOException {
+        BibteXMLImporter importer = new BibteXMLImporter();
+
+        List<String> list = Arrays.asList("BibteXMLImporter.xml", "BibteXMLImporterEmpty.xml");
+
+        for (String str : list) {
+            try (InputStream is = BibteXMLImporter.class.getResourceAsStream(str)) {
+                Assert.assertTrue(importer.isRecognizedFormat(is));
+            }
+        }
+    }
+
+    @Test
+    public void testIsRecognizedFormatReject() throws IOException {
+        BibteXMLImporter importer = new BibteXMLImporter();
+
+        List<String> list = Arrays.asList("oai2.xml", "oai22.xml", "oai23.xml");
+
+        for (String str : list) {
+            try (InputStream is = BibteXMLImporter.class.getResourceAsStream(str)) {
+                Assert.assertFalse(importer.isRecognizedFormat(is));
+            }
+        }
+    }
+
+    @Test
+    public void testImportEntries() throws IOException {
+
+        BibteXMLImporter importer = new BibteXMLImporter();
+
+        try (InputStream is = BibteXMLImporter.class.getResourceAsStream("BibteXMLImporter.xml")) {
+            List<BibEntry> entries = importer.importEntries(is, new OutputPrinterToNull());
+
+            BibEntry entry = entries.get(0);
+            Assert.assertEquals(entry.getCiteKey(), "");
+            Assert.assertEquals("article", entry.getType());
+            Assert.assertEquals("Test Article", entry.getField("article"));
```

Comment:
Just now I realise that there were a few changes since I merged with master. 
Because before that I could read with getFieldNames() and getOptionalFieldNames() all available fields for any kind of entry. In this case every entry had the same fields except the one field. The field named after the type. So I used it to see whether JabRef can read the fields. Still when I use getFieldNames() I get the fields "article" and "bibtexkey". getOptionalFieldNames() was removed. So in this case it´s just to test if the value is readable.